agent_build/tools/build_in_docker/__init__.py

```diff
@@ -0,0 +1,140 @@
+    Just a simple wrapper around "docker build" command.
+    :param architecture: Architecture of the build. Translated to the docker's --platform option.
+    :param image_name: Name of the result image.
+    :param dockerfile_path: Path to the Dockerfile.
+    :param build_context_path: Build context path.
+    :param build_args: Dockerfile build arguments (--build-arg).
+    """
+
+    # Enable docker Buildkit.
+    env = os.environ.copy()
+    env["DOCKER_BUILDKIT"] = "1"
+
+    build_args = build_args or {}
+
+    build_arg_options = []
+
+    for name, value in build_args.items():
+        build_arg_options.append("--build-arg")
+        build_arg_options.append(f"{name}={value}")
+
+    subprocess.check_call(
+        [
+            "docker",
+            "build",
+            "--platform",
+            architecture.as_docker_platform.value,
+            "-t",
+            image_name,
+            *build_arg_options,
+            "--label",
+            "scalyr-agent-build",
+            "-f",
+            str(dockerfile_path),
+            str(build_context_path),
+        ],
+        env=env,
+    )
+
+
+def build_stage(
+    command: str,
+    stage_name: str,
+    architecture: constants.Architecture,
+    image_name: str,
+    base_image_name: str,
+    output_path_mappings: Dict[pl.Path, pl.Path] = None,
+):
+    """
+    Run a custom command using the 'docker build' by specifying special Dockerfile which is located
+        in the same directory with current module.
+    :param command: Command to execute.
+    :param stage_name: Name of one of the stages in the Dockerfile. Those stages are used in different scenarios, for
+        example agent package build or package tests, and include different set of files needed for those scenarios.
+    :param architecture: Architecture of the build. Translated to the docker's '--platform' option.
+    :param image_name: Name of the result image.
+    :param base_image_name: Name of the base image to use.
+    :param output_path_mappings: Dict with mapping of the host paths to the paths that are used inside the docker.
+        After the build is completed, files or folders that are located inside the docker will be copied to the
+        appropriate host paths.
+    """
+    run_docker_build(
+        architecture=architecture,
+        image_name=image_name,
+        dockerfile_path=__PARENT_DIR__ / "Dockerfile",
+        build_context_path=__SOURCE_ROOT__,
+        build_args={
+            "BASE_IMAGE_NAME": base_image_name,
+            "COMMAND": command,
+            "BUILD_STAGE": stage_name,
+        },
+    )
+
+    # If there are output mapping specified, than we has to copy result files from the result build.
+    # To do that we have to create a container and copy files from it.
+    if output_path_mappings:
+        container_name = image_name
+        # Remove the container with the same name if exists.
+        subprocess.check_call(["docker", "rm", "-f", container_name])
```

Comment:
To make debugging and troubleshooting a bit easier, do we also want to add some logging and print which commands are ran? Since, IIRC (please correct me if I'm wrong, ``check_call()`` doesn't print which command it runs).

Reply:
Added wrapper for the `subprocess.check_call` which logs its command.